orderer/common/types/content.go

```diff
@@ -0,0 +1,14 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package types
+
+// JoinBody is encoded in the body of a Join POST request when Content-Type is application/json.
+type JoinBody struct {
```

Comment:
Why do you need this?

Reply:
This task implements POST with a body of type application/json, with structure:
```
{
   configBlock: "block content (binary, after marshaling to protobuf) in base64"
}
```
this structure helps marshal and unmarshal a []byte slice that contains the block bytes.

Support for Content-type application/octet-stream and multipart/form-data will be added later. The plan is to support all 3 types, see:
https://jira.hyperledger.org/browse/FAB-17891

I am starting with this one because it is easier to unit-test & integration-test (IMO), but eventually all 3 will be supported, as planed.

Reply:
I do not think there is a need in augmenting binary into json as I also do not see particular reason to support all three.

Reply:
I added support for multipart/form-data.

Reply:
in such case you do not need this `JoinBody` struct, right?

Reply:
I would like to keep both `multipart/form-data` & `application/json`.

Reply:
I do not think we need to introduce two methods to achieve same goal, hence I think we better to start with one option only.

Reply:
OK. It is easy enough to add if someone needs it.